rpc/src/services/dev_services.rs

```diff
@@ -910,16 +914,79 @@ pub(crate) async fn get_shell_automaton_endorsing_rights(
     Ok(response)
 }
 
+fn adjust_times(base_time: u64, value: &mut serde_json::Value) -> anyhow::Result<()> {
+    let obj = value
+        .as_object_mut()
+        .ok_or(anyhow::format_err!("object expected"))?;
+    obj.iter_mut().for_each(|(_, v)| {
+        if let Some(v) = v.as_object_mut() {
+            v.iter_mut().for_each(|(k, v)| {
+                if k.ends_with("_time") {
+                    if let Some(t) = v.as_i64() {
+                        *v = serde_json::json!(t - (base_time as i64));
+                    }
+                }
+            })
+        }
+    });
+    Ok(())
+}
+
 pub(crate) async fn get_shell_automaton_endorsements_status(
     block_hash: Option<BlockHash>,
+    payload_hash: Option<BlockPayloadHash>,
+    level: Option<i32>,
+    round: Option<i32>,
+    base_time: Option<u64>,
     env: &RpcServiceEnvironment,
 ) -> anyhow::Result<serde_json::Value> {
+    use shell_automaton::mempool::mempool_actions::ConsensusOperationMatcher;
+    let matcher = match (block_hash, payload_hash, level, round) {
+        (Some(block_hash), None, None, None) => {
+            ConsensusOperationMatcher::endorsement_branch(block_hash)
+        }
+        //        (None, Some(payload_hash), None, None) => ConsensusOperationMatcher::endorsement_payload(payload_hash),
```

Comment:
what about this commented out case? not needed?

Reply:
Now it is not needed, a stub for the future.